javascript/src/main/components/BasicCourseSearch/BasicCourseTable.js

```diff
@@ -1,23 +1,113 @@
+        (section) => {
+          if (section.section % 100 != 0){
+            numSections++;
+          }
+          if(course.classSections.length <= 1 || (section.section % 100 == 0 && numSections == 0)) {
+            section.course =
+            {
+              courseId: course.courseId,
+              title: course.title,
+              unitsFixed: course.unitsFixed,
+              noSections: "true"
+            };
+          } else {
+            section.course =
+            {
+              courseId: course.courseId,
+              title: course.title,
+              unitsFixed: course.unitsFixed,
+              noSections: "false"
+            };
+          }
+          if (section.section % 100 == 0){
+            numSections = 0;
+          }
+         sections.push(section);
+       }
+      )
+    }
+  );
+  sections.reverse();
   
+const rowStyle = (row, rowIndex) => {
+  return (row.section % 100 == 0)? {backgroundColor: '#CEDEFA'}: (rowIndex % 2 == 1)? {backgroundColor: '#EDF3FE'}: {backgroundColor: '#FFFFFF'}
+}
+const dataAlignment = (cell, row) => {
+  const alignmnet = (row.section % 100 == 0)? 'left': 'right';
+  return alignmnet
+}
+const renderCourseId = (cell, row) => {
+  const courseId = (row.section % 100 == 0)? row.course.courseId: "";
+  return ( courseId )
+}
+const renderCourseTitle = (cell, row) => {
+  const courseTitle = (row.section % 100 == 0)? row.course.title: "";
+  return (  courseTitle )
+}
+const renderInstructors = (cell, row) => {
+  const instructor = (row.instructors.length > 0)? row.instructors[0].instructor: "TBD";
+  return ( instructor )
+}
+const renderAddCourseButton = (id,cell,row) => {
+  const button = (row.section % 100 != 0 || row.course.noSections == "true")? 
+    <Button data-testid="add-course-button" onClick={() => { 
+      history.push(``); 
+      console.log('To Be Implemented')}}>Add Course</Button>: "";
+  return (
+      button
+  )
+}
   const columns = [{
-    dataField: 'courseId',
-    text: 'Course Number'
+    dataField: 'course.courseId',
+    text: 'Course Number',
+    isDummyField: true,
+    formatter: (cell, row) => renderCourseId(cell, row)
+  },{
+    dataField: 'course.title',
+    text: 'Title',
+    isDummyField: true,
+    formatter: (cell, row) => renderCourseTitle(cell, row)
+  },{
+    dataField: 'section',
+    text: 'Section',
+    align: (cell, row) => dataAlignment(cell, row)
   },{
-    dataField: 'title',
-    text: 'Title'
+    dataField: "instructors",
+    text: "Instructor",
+    isDummyField: true,
+    formatter: (cell, row) => renderInstructors(cell, row),
+    align: (cell, row) => dataAlignment(cell, row)
   },{
-    dataField: 'unitsFixed',
-    text: 'Unit'
+    dataField: 'enrollCode',
+    text: 'Enroll Code',
+    align: (cell, row) => dataAlignment(cell, row)
+  },{
+    dataField: 'course.unitsFixed',
+    text: 'Unit',
+    align: (cell, row) => dataAlignment(cell, row)
   }
 ];
-  
+
+  columns.push({
```

Comment:
So, I think we should only be doing this push if the user is a "member".  That is, we should check to see if they are logged in, and if they are, if they are a member.  Otherwise, the Add Course button should not appear.

Is that something you can fix?

Reply:
We added a check here which works so this should be all set.